back/src/companies/companyUtils.ts

```diff
@@ -0,0 +1,65 @@
+import { searchCompany } from "./search";
+import { splitAddress } from "../common/addresses";
+import { Company } from "@prisma/client";
+import { isDefinedStrict } from "../common/helpers";
+
+/**
+ * Retourne l'adresse splittée d'une entreprise ('street', 'postalCode', 'city', 'country').
+ *
+ * Essaie d'abord d'interroger les APIs / ES pour avoir l'adresse splittée,
+ * et en cas de souci, essaie de splitter l'adresse manuellement.
+ *
+ * Si le split n'a pas fonctionné, retourne tous les champs à null.
+ *
+ * Attention: certaines entreprises ont des addresses du genre "codePostal ville",
+ * auquel cas on retourne "" pour la rue.
+ */
+export const getCompanySplittedAddress = async (company: Company) => {
+  let searchedCompany;
```

Comment:
```suggestion
let searchedCompany: CompanySearchResult| null = null;
```
te permettrait ici de ne pas perdre le typage ici.